### ISSUE TITLE
Add a better newrelic_account data source

### ISSUE THREAD

**Issue by alisson276** [newrelic/terraform-provider-newrelic#1982]
### Feature Description
The current [newrelic_account data source](https://registry.terraform.io/providers/newrelic/newrelic/latest/docs/data-sources/account) is quite limited, we cannot get the current account_id (the one specified in the provider section) because or account_id is required (what is confusing for me, if I know the account_id I don't need a data source, I can set it as a local var, for example) or account name. There is no option to get the current selected accounted ID from the provider (without parameters).

### Describe Alternatives
A good example of this feature you can find in the [AWS provider](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/caller_identity). An usage example would be:
```
terraform {
  required_providers {
    newrelic = {
      source  = "newrelic/newrelic"
      version = "~> 3.0"
    }
}

provider "newrelic" {
  account_id = 1234567
  region = "US"
}

data "newrelic_account" "current" {}

resource "newrelic_alert_policy" "this" {
  account_id          = data.newrelic_account.current.account_id
  name                = "Example name"
}
```

### Additional context
I'm requesting this because I'm facing a problem, which is, if I don't set an account_id on newrelic_alert_policy resource, it tries to recreate the resource every time I run a plan.

**Comment by tguvdamm** [comment]
I believe this is just a bug in the alert_policy. It should inherit the account_id and not recreate the resource. I'm also bumping into this.

**Comment by pranav-new-relic** [comment]
Hi, this feature/fix has been released with the latest version of Terraform, [v3.22.0](https://github.com/newrelic/terraform-provider-newrelic/releases/tag/v3.22.0). Closing the issue, thank you for reporting.
